### DISCUSSION TITLE
Ideas for using Hydrogen as a LV2 plugin

### DISCUSSION THREAD

**Discussion by mauser** [hydrogen-music/hydrogen#1026] [Hydrogen on Linux]
Hi everyone,

i'm currently (well, from time to time...) working on a version of Hydrogen which works as a LV2 plugin. Currently, it works as a simple sampler which can load a drumkit, has midi input and a stereo output. Which other features would be interesting?

**Comment by trebmuh** [comment]
As a first thought, a multi output (such as the "create per-instrument output") would be great.

**Comment by dingodoppelt** [comment]
I'm really looking forward to using hydrogen as an lv2 plugin, great idea and thanks in advance :)
I'd like access to the mixer, at least the master fader. It seems it isn't saved in hydrogen and I have to set it manually every time.
I couldn't find the code though. Is it ready to be tested yet, as I would love to do so?
cheers, nils

**Reply by mauser** [reply]
Hi Nils! 

It is not quite ready yet. I have already implemented the core of the plugin, but i now i need to work on the UI and the communication between the UI and the core (well, "working" means "shamelessly copying from the lv2 sampler example plugin"). I've started to work again on the topic today and i hope that there will be some progress in the next two weeks.. I'm going to give you an update when the feature can be tested!

**Comment by funkypou** [comment]
Hi mauser,

Thanks for what you're doing. In my opinion, you should focus on what makes Hydrogen so useful and unique and can't be found in other lv2 plugins. There are other lv2 drum sampler plugins like AVL Drumkits, DrumGizmo or DrMr.

I think what makes Hydrogen so special and efficient is its sequencer divided in 2 parts : pattern and song editors. It allows fast creation of a whole drum tracks. That's the kind of feature that is lacking in Ardour, on which creating a drum track is quite painful and takes time. I don't know how a sequencer plugin could integrate Ardour's workflow but if it's possible, it would be amazing and appreciated by many users.

**Comment by dingodoppelt** [comment]
Are there maybe any updates on this? I just ordered a raspberry pi zero 2 w and love the idea of it being my future mobile lv2 host and hydrogen (as awesome as it is!!) is certainly the first thing to install to get my fingerdrums on the road ;)
